### PR TITLE
Add support for kubernetes host detection

### LINKED ISSUE #4456: Update useragent string to indicate where minio was installed on
This would be very helpful for debugging and supportability.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add support for kubernetes host detection.

Additionally improve what we print for `docker pull`
such that it is precisely the relevant release tag.

## Motivation and Context
Fixes #4456 

## How Has This Been Tested?
Manually using kubernetes deployment. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/update-main.go [hunk 1]
```diff
@@ -135,6 +146,9 @@ func getUserAgent(mode string) string {
 	if mode != "" {
```
Comment (balamurugana) [inline cmd/update-main.go:147]: please update comment of this function to indicate `; kubernetes`

File: cmd/update-main.go [hunk 2]
```diff
@@ -226,11 +240,36 @@ func getLatestReleaseTime(timeout time.Duration, mode string) (releaseTime time.
...
+	relTag = strings.Replace(relTag, ":", "-", -1)
+	relTag = strings.Replace(relTag, ",", "", -1)
+	if relPrefix != "" {
+		return relPrefix + "." + relTag
+	}
+	return relTag
+}
+
+func getDownloadURL(buildDate time.Time) (downloadURL string) {
+	rTag := releaseTag(buildDate.Format(time.RFC3339), "RELEASE")
```
Comment (balamurugana) [inline cmd/update-main.go]: you could use `minioReleaseTagTimeLayout`.  IMO `releaseTag()` is not required as separate function.

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: cmd/update-main.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/update-main.go]: Ah yes. Forgot about it.

File: cmd/update-main_test.go [hunk 3]
```diff
@@ -30,6 +30,27 @@ import (
...
 
+func TestDownloadURL(t *testing.T) {
+	minioVersion1 := UTCNow()
+	durl := getDownloadURL(minioVersion1)
+	if runtime.GOOS == "windows" {
+		if durl != minioReleaseURL+"minio.exe" {
+			t.Errorf("Expected %s, got %s", minioReleaseURL+"minio.exe", durl)
+		}
+	}
+	if durl != minioReleaseURL+"minio" {
```
Comment (krisis) [inline cmd/update-main_test.go]: This is why the appveyor build failed. This check should not run on windows.

File: cmd/update-main_test.go [hunk 4]
```diff
@@ -136,6 +157,21 @@ func TestGetCurrentReleaseTime(t *testing.T) {
...
+func TestIsKubernetes(t *testing.T) {
+	os.Setenv("KUBERNETES_SERVICE_HOST", "10.11.148.5")
+	kubernetes := IsKubernetes()
+	if !kubernetes {
+		t.Fatalf("We should be running in kubernetes Expected %t, got %t", true, kubernetes)
+	}
+	os.Unsetenv("KUBERNETES_SERVICE_HOST")
+	kubernetes = IsKubernetes()
+	if kubernetes {
+		t.Fatalf("We should be running in kubernetes Expected %t, got %t", false, kubernetes)
```
Comment (krisis) [inline cmd/update-main_test.go]: s/We should be/ We shouldn't be/

**Review by krisis — CHANGES_REQUESTED** [review]

File: cmd/update-main.go [hunk 5]
```diff
@@ -226,11 +241,25 @@ func getLatestReleaseTime(timeout time.Duration, mode string) (releaseTime time.
 	return parseReleaseData(data)
 }
 
-func getDownloadURL() (downloadURL string) {
+// Kubernetes deploy doc link.
+var kubernetesDeploymentDoc = "https://docs.minio.io/docs/deploy-minio-on-kubernetes"
```
Comment (balamurugana) [inline cmd/update-main.go]: Why is it `var`?

File: cmd/update-main.go [hunk 6]
```diff
@@ -226,11 +241,25 @@ func getLatestReleaseTime(timeout time.Duration, mode string) (releaseTime time.
...
+
+func getDownloadURL(buildDate time.Time) (downloadURL string) {
+	// Construct release tag name.
+	rTag := "RELEASE." + buildDate.Format(minioReleaseTagTimeLayout)
```
Comment (balamurugana) [inline cmd/update-main.go]: I see `rTag` is applicable only to when `IsDocker()`.  You would need to move to `if` block.

File: cmd/update-main_test.go [hunk 7]
```diff
@@ -136,6 +158,21 @@ func TestGetCurrentReleaseTime(t *testing.T) {
 	}
 }
 
+// Tests if the environment we are running is in kubernetes.
+func TestIsKubernetes(t *testing.T) {
+	os.Setenv("KUBERNETES_SERVICE_HOST", "10.11.148.5")
+	kubernetes := IsKubernetes()
+	if !kubernetes {
+		t.Fatalf("We should be running in kubernetes. Expected %t, got %t", true, kubernetes)
```
Comment (balamurugana) [inline cmd/update-main_test.go]: The text `We should be running in kubernetes` doesn't convey the message clearly.  I think `Expected... Got...` is good enough.  But its your call.

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: cmd/update-main.go (hunk 5 above)
Comment (harshavardhana) [reply cmd/update-main.go]: Done

File: cmd/update-main.go (hunk 6 above)
Comment (harshavardhana) [reply cmd/update-main.go]: Done

File: cmd/update-main_test.go (hunk 7 above)
Comment (harshavardhana) [reply cmd/update-main_test.go]: Done

**Review by balamurugana — APPROVED** [review]

**Review by krisis — APPROVED** [review]

**Pull request merged**